tests/integration/security/sds_ingress/ingress_test.go

```diff
@@ -172,7 +172,69 @@ func TestSingleMTLSGateway_CompoundSecretRotation(t *testing.T) {
+			defer ingressutil.DeleteIngressKubeSecret(t, ctx, credName)
+
+			ns := ingressutil.SetupTest(ctx)
+			ingressutil.SetupConfig(t, ctx, ns, ingressutil.TestConfig{
+				Mode:           "MUTUAL",
+				CredentialName: credName[0],
+				Host:           host,
+			})
+			// Wait for ingress gateway to fetch key/cert from Gateway agent via SDS.
+			ing := ingress.NewOrFail(t, ctx, ingress.Config{Istio: inst})
+			tlsContext := ingressutil.TLSContext{
+				CaCert:     ingressutil.CaCertA,
+				PrivateKey: ingressutil.TLSClientKeyA,
+				Cert:       ingressutil.TLSClientCertA,
+			}
+			err := ingressutil.SendRequest(ing, host, credName[0], ingress.Mtls, tlsContext,
+				ingressutil.ExpectedResponse{ResponseCode: 200, ErrorMessage: ""}, t)
+			if err != nil {
+				t.Fatalf("unable to retrieve 200 from product page at host %s: %v", host, err)
+			}
+
+			// key/cert rotation
+			ingressutil.RotateSecrets(t, ctx, credName, ingress.Mtls, ingressutil.IngressCredentialB)
+			// Use old server CA cert to set up SSL connection would fail.
+			err = ingressutil.SendRequest(ing, host, credName[0], ingress.Mtls, tlsContext,
+				ingressutil.ExpectedResponse{ResponseCode: 0, ErrorMessage: "certificate signed by unknown authority"}, t)
+			if err != nil {
+				t.Fatalf("unable to retrieve 404 from product page at host %s: %v", host, err)
+			}
+
+			// Use new server CA cert to set up SSL connection.
+			tlsContext = ingressutil.TLSContext{
+				CaCert:     ingressutil.CaCertB,
+				PrivateKey: ingressutil.TLSClientKeyB,
+				Cert:       ingressutil.TLSClientCertB,
+			}
+			err = ingressutil.SendRequest(ing, host, credName[0], ingress.Mtls, tlsContext,
+				ingressutil.ExpectedResponse{ResponseCode: 200, ErrorMessage: ""}, t)
+			if err != nil {
+				t.Fatalf("unable to retrieve 200 from product page at host %s: %v", host, err)
+			}
+		})
+}
+
+// TestSingleMTLSGateway_CompoundSecretRotation tests a single mTLS ingress gateway with SDS enabled and use the tls cert instead of generic cert
+// Verifies behavior in these scenarios.
+// (1) A valid kubernetes secret with key/cert and client CA cert is added, verifies that SSL connection
+// termination is working properly. This secret is a compound secret.
+// (2) After key/cert rotation, client needs to pick new CA cert to complete SSL connection. Old CA
+// cert will cause the SSL connection fail.
+func TestSingleMTLSGatewayWithCaCert_CompoundSecretRotation(t *testing.T) {
+	framework.
+		NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			var (
+				credName = []string{"testsinglemtlsgatewaywithcacert-compoundsecretrotation"}
+				host     = "testsinglemtlsgatewaywithcacert-compoundsecretrotation.example.com"
+			)
+
+			// Add kubernetes secret to provision key/cert for ingress gateway.
+			ingressutil.CreateIngressKubeSecret(t, ctx, credName, ingress.Mtls, ingressutil.IngressCredentialA, true)
```

Comment:
isCompoundGeneric should be false in this case, is it?

Reply:
Yes it should be a false here. However as I change the flag from isCompoundGeneric to isCompoundWithCaCert to specify the integration test I added. I will leave it true. And all other function will use the parameter as false.